olclient/openlibrary.py

```diff
@@ -656,6 +656,49 @@ def save(self, comment):
                 url = self.OL.base_url + '/authors/%s.json' % self.olid
                 return self.OL.session.put(url, json.dumps(body))
 
+            def works(self, limit=50, offset=0):
+                """Returns a list of OpenLibrary Works associated with an OpenLibrary Author.
+
+                Args:
+                    olid (unicode) - OpenLibrary ID for author to search within
+                                    Open Library's database of authors to retrieve his Works.
+                    name (unicode) - name of an Author to search for within OpenLibrary.
+                    limit (integer) - number of Author's Works to return.
+                    offset (integer) - offset number to aid pagination.
+                Returns:
+                    A (list) of Works from the OpenLibrary associated with the
+                    Author.
+
+                Usage:
+                    >>> from olclient.openlibrary import OpenLibrary
+                    >>> ol = OpenLibrary()
+                    >>> ol.Author.get('OL39307A').works()
+                    or
+                    >>> ol.Author.get('OL39307A').works(limit=20)# to obtain the first 20 works of the author
+                    >>> ol.Author.get('OL39307A').works(limit=20, offset=20)# to obtain the next 20 works of the author
+                    or
+                    >>> author_obj = ol.Author.get(ol.Author.get_olid_by_name('Dan Brown'))
+                    >>> author_obj.works()
+                    or
+                    >>> ol.Author.get(ol.Author.get_olid_by_name('Dan Brown')).works()
+                """
+                path = '/authors/%s/works.json' % self.olid
+
+                if limit:
+                    # by default the limit is set in the function definition
+                    path += '/?limit=%s' % limit
+
+                if offset:
+                    # by default the offset is set in the function definition
+                    path += '&offset=%s' % offset
+
+                try:
+                    response = self.OL._get_ol_response(path)
+                    return response.json()
+                except Exception as e:
```

Comment:
I know we probably have tons of places in our code where we `except Exception` -- is there an opportunity here to do a better job of catching a specific type of exception?

If you really want to punt on this, let's open a followup issue: "More targeted exception handling for `Author.works`

Reply:
@mek,
It would good if we open a dedicated issue to update the error handling alone.
As I have found the error handling originating from a different section of the code which will require further probing and an overall correction.

Reply:
A dedicated Issue #139 to improve overall error handling is raised.